### ISSUE TITLE
0.19

### ISSUE THREAD

**Issue by jxxcarlson** [elm-community/string-extra#27]
Hello, the 0.19 alpha is out:

   https://gist.github.com/evancz/8e89512dfa9f68903f05f1ac4c44861b

   -- Evan has asked for a quiet no-blogging period with discussion restricted th the elm-dev Slack channel

   https://alpha.elm-lang.org/  --- a list of updated packages

Any chance of updating string-extra to 0.19 soon?

Thanks so much!!

**Comment by ianmackenzie** [comment]
As part of the upgrade process, the `toCodePoints` and `fromCodePoints` functions should probably just be removed. (I say that as the person who contributed those functions in the first place!) They shouldn't be necessary any more now that the core `Char` type actually does correspond to a single Unicode code point (instead of a UTF-16 code unit); `toCodePoints` can now be replaced by

```elm
String.toList >> List.map Char.toCode
```

and `fromCodePoints` can be replaced by

```elm
List.map Char.fromCode >> String.fromList
```

Alternatively, to avoid breaking any user code, the implementations of `toCodePoints` and `fromCodePoints` could be changed to use the above implementations. Either way, they should *not* be left as is, because their implementations assume `Char` values are 16-bit code units - as a result, at least `toCodePoints` is broken in Elm 0.19.

**Comment by jaapz** [comment]
@ianmackenzie i think it'd be nice to have those implementations of toCodePoints and fromCodePoints in the 0.19-compatible release of string-extras, so we don't have to do a major version bump. @lorenzo what do you think?

**Comment by lorenzo** [comment]
Can we actually have a version compatible with both 0.18 and 0.19? If that is possible, then it would be desirable

**Comment by ianmackenzie** [comment]
I don't see an easy way to have a version that's compatible with both 0.18 and 0.19, due to the fundamental differences in `Char` representation. I can only think of really messy solutions like doing a runtime "fingerprinting" check of some sort to figure out which version we're on, and changing the logic accordingly.

**Comment by lorenzo** [comment]
@ianmackenzie we don't need to make it compatible with both versions, just one and we keep separate branches in case we need to make fixes to 0.18.

**Comment by ianmackenzie** [comment]
Just opened #30 to update the conversion functions.
